components/kyma-environment-broker/internal/config/reader_test.go

```diff
@@ -0,0 +1,239 @@
+			},
+		},
+	}
+	cfgMap, err := fixConfigMap()
+	require.NoError(t, err)
+
+	mockK8sClient := mockK8sClient{}
+	fakeK8sClient := fake.NewClientBuilder().Build()
+	logger := logrus.New()
+	logger.SetFormatter(&logrus.JSONFormatter{})
+
+	t.Run("should return error while fetching configmap on List() of K8s client", func(t *testing.T) {
+		// given
+		cfgReader := config.NewConfigMapReader(ctx, mockK8sClient, logger)
+
+		// when
+		rawCfg, err := cfgReader.Read(kymaVersion, broker.AzurePlanName)
+
+		// then
+		require.Error(t, err)
+		logger.Error(err)
+		assert.Equal(t, "", rawCfg)
+	})
+
+	t.Run("should return error while verifying configuration configmap existence", func(t *testing.T) {
+		// given
+		cfgReader := config.NewConfigMapReader(ctx, fakeK8sClient, logger)
+
+		// when
+		rawCfg, err := cfgReader.Read(kymaVersion, broker.AzurePlanName)
+
+		// then
+		require.Error(t, err)
+		logger.Error(err)
+		assert.Equal(t, "", rawCfg)
+
+		// given
+		err = fakeK8sClient.Create(ctx, cfgMap)
+		require.NoError(t, err)
+
+		err = fakeK8sClient.Create(ctx, redundantCfgMap)
+		require.NoError(t, err)
+
+		// when
+		rawCfg, err = cfgReader.Read(kymaVersion, broker.AzurePlanName)
+
+		// then
+		require.Error(t, err)
+		logger.Error(err)
+		assert.Equal(t, "", rawCfg)
+	})
+
+	t.Run("should return error while getting config string for a plan", func(t *testing.T) {
+		// given
+		err = fakeK8sClient.Delete(ctx, cfgMap)
+		require.NoError(t, err)
+
+		cfgReader := config.NewConfigMapReader(ctx, fakeK8sClient, logger)
+
+		// when
+		rawCfg, err := cfgReader.Read(kymaVersion, broker.AzurePlanName)
+
+		// then
+		require.Error(t, err)
+		logger.Error(err)
+		assert.Equal(t, "", rawCfg)
+	})
+}
+
+func fixConfigMap() (*coreV1.ConfigMap, error) {
+	yamlFilePath := path.Join("testdata", kebConfigYaml)
+	contents, err := os.ReadFile(yamlFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("while reading configmap")
+	}
+
+	var tempCfgMap tempConfigMap
+	err = yaml.Unmarshal(contents, &tempCfgMap)
+	if err != nil {
+		return nil, fmt.Errorf("while unmarshalling configmap")
+	}
+
+	return tempCfgMap.toConfigMap(), nil
+}
+
+type tempConfigMap struct {
+	APIVersion string            `yaml:"apiVersion,omitempty"`
+	Kind       string            `yaml:"kind,omitempty"`
+	Metadata   tempMetadata      `yaml:"metadata,omitempty"`
+	Data       map[string]string `yaml:"data,omitempty"`
+}
+
+type tempMetadata struct {
+	Name      string            `yaml:"name,omitempty"`
+	Namespace string            `yaml:"namespace,omitempty"`
+	Labels    map[string]string `yaml:"labels,omitempty"`
+}
+
+func (m *tempConfigMap) toConfigMap() *coreV1.ConfigMap {
+	return &coreV1.ConfigMap{
+		TypeMeta: metaV1.TypeMeta{
+			Kind:       m.Kind,
+			APIVersion: m.APIVersion,
+		},
+		ObjectMeta: metaV1.ObjectMeta{
+			Name:      m.Metadata.Name,
+			Namespace: m.Metadata.Namespace,
+			Labels:    m.Metadata.Labels,
+		},
+		Data: m.Data,
+	}
+}
+
+type mockK8sClient struct{}
```

Comment:
I'd name it like: "failingK8sClient" or something like this which tells us it always fails

Reply:
I will rename this fake client to "failingK8sClient"